### ISSUE TITLE
Change the CSS of all the samples use canvas { display: block; }

### ISSUE THREAD

**Issue by greggman** [mrdoob/three.js#16523]
##### Description of the problem

Most of the samples are using CSS incorrectly I believe. They do this

```
		body {
			margin: 0px;
			overflow: hidden;
		}
```

What they should do is

```
		body {
			margin: 0;
		}
		canvas {
			display: block;
		}
```

The only reason for `overflow:hidden` as far as I can tell is because things scroll a little if you don't add it. But the only reason things scroll a little is because the default type for `<canvas>` is `display: span;`.  Span adds some padding at the end which is why the extra scrolling appears. 

Setting the canvas to `display: block`  fixes it.

See https://jsfiddle.net/greggman/aL45pmkt/ to see the padding

Also margin should probably just be `0` not `0px`

Here's a three.js example

https://jsfiddle.net/greggman/fsLq9b6w/

Comment out the `canvas { display: block; }` and you'll get the scrollbar that `overflow: hidden` was there to prevent. But overflow hidden is just hiding the actual bug which is that the canvas wasn't set to display block; In other words it's hiding the buggy extra padding. 

Would it be better to fix the bug (the extra padding) than leave the bug and continue to hack around it with overflow hidden?

I see this was asked before: https://github.com/mrdoob/three.js/issues/14751

##### Three.js version

- [x] Dev

##### Browser

- [x] All of them

##### OS

- [x] All of them

**Comment by mrdoob** [comment]
Yes please! I learnt this while battling with all the controls issues.

**Comment by sciecode** [comment]
Doesn't work for me. [example](https://rawcdn.githack.com/sciecode/three.js/e7b16ec58c2fef7d93e98f0f2e3c759f1aace3ca/examples/webgl_animation_cloth.html) - [code](https://github.com/sciecode/three.js/commit/e7b16ec58c2fef7d93e98f0f2e3c759f1aace3ca)


![image](https://user-images.githubusercontent.com/3927951/58254209-9e604900-7d40-11e9-8852-0088e7c879df.png)

**Comment by mrdoob** [comment]
> **edit**: setting div `box-sizing` to `border-box` solves this. which leads me to believe the problem here is that, in the examples, the canvas needs to be appended to a div container. 

Close but not quite. The problem is the div#info which has `width: 100%; padding: 10px;` and forced the div to be bigger than the viewport. `box-sizing: border-box` solves that.

It has taken me 2 decades messing with CSS to understand this 😂

**Comment by sciecode** [comment]
> Close but not quite. The problem is the div#info which has `width: 100%; padding: 10px;` and forces the div to be bigger than the viewport. `box-sizing: border-box` solves that.

That, indeed, makes a lot more sense 😂 
Also explains why removing container's margin and padding didn't affect the problem.

In that case, its a simple matter of choosing between removing horizontal padding `padding:  10px 0;` or setting  to `box-sizing: border-box;` on div#info.

**Comment by greggman** [comment]
Another issue (which is also buggy)

**on desktop** Chrome and Safari, `overflow: hidden` prevents scroll bounce, the thing where the browser scrolls a little past the end of the page and then bounces back. Firefox doesn't seem to have scroll bounce at all so no issue there.

**on mobile** scroll bounce is not prevented.

To prevent scroll bounce on desktop is

```
windowOrSomeElement.addEventListener('wheel', (event) => {
  event.preventDefault();
}, {passive: false});
```

To prevent scroll bounce on mobile it's

```
windowOrSomeElement.addEventListener('touchstart', (e) => {
  e.preventDefault();
}, {passive: false});
```

To test

[This page](http://twgljs.org/examples/tiny.html) uses `canvas: { display: block;} `. It scroll bounces every where

[This page](https://threejs.org/examples/webgl_geometry_colors.html) uses `overflow:hidden` so no scroll bounce on desktop but does bounce on iOS safari

[This page](https://greggman.github.io/doodles/x-ray.html) uses `preventDefault` on both wheel and touchmove. It doesn't bounce at all

Whether you want to add that to any of the examples is up to you. Given they all bounce on mobile it probably doesn't matter they start bouncing on desktop if you take out the `overflow: hidden` but you thought you should be aware before you do it.

**Comment by mrdoob** [comment]
Oh yeah, I remember adding [this hack](https://github.com/mrdoob/three.js/blob/dev/editor/index.html#L11) to the editor just for this...

**Comment by greggman** [comment]
[that hack](https://github.com/mrdoob/three.js/blob/dev/editor/index.html#L11) doesn't work on iOS

One more solution to scroll bounce

```
body {
      position: fixed;
      overflow: hidden;
}
```

appears to also work everywhere. [Example](https://greggman.github.io/doodles/tween.html)

just passing it on.

**Comment by mrdoob** [comment]
Oh dear...

I'm thinking it may be okay to have the bounce thing. That way the content behaves like the user "expects".

It'll then be up to the controls to override the behaviour. I guess that'll also make it easier to know when a page is using controls or not.

**Comment by gkjohnson** [comment]
I'm actually still seeing scroll bars on the examples pages on my windows surface laptop but not my mac laptop.

One difference I noticed is that one has a device pixel ratio with a fractional component -- the windows laptop has a devicePixelRatio of `1.25` while the mac reports `2` but I'm not sure why that would cause problems...

**Comment by mrdoob** [comment]
Interesting. I'll have to try on a Surface.
If anything, this change now exposed this issue.

**Comment by mrdoob** [comment]
@gkjohnson My surface laptop pixel ratio is `2`. I tried zooming out Chrome so the so `devicePixelRatio` has weird values like `1.600000023841858` and I still don't see scrollbars.

What examples do you get scrollbars with? Is that on Edge? Chrome:? ...

**Comment by gkjohnson** [comment]
> What examples do you get scrollbars with? Is that on Edge? Chrome:? ...

Every example (that I clicked through) in Firefox, Edge, and Chrome 😵

> My surface laptop pixel ratio is 2. I tried zooming out Chrome so the so devicePixelRatio has weird values like 1.600000023841858 and I still don't see scrollbars.

I just realized this depends on the windows DPI scaling setting ([here](https://www.lifewire.com/using-windows-10-display-scaling-4587328)). I set my scaling to 125%. It sounds like yours is set to 200%. If I change my setting to 200% I can no longer replicate the problem and my `devicePixelRatio` is set to 2. See if you can replicate it with 125%? The issue doesn't always happen -- if the scrollbars don't show up try resizing the window and maybe refreshing. While resizing the window I see scroll bars flickering in and out.

And just for a little more information my Surface Laptop is the 2017 version with i5 and 2256 x 1504 resolution.

I tried fiddling around with a few settings to see if I could fix it but `overflow: hidden` was what I found to work.

**Comment by greggman** [comment]
@gkjohnson, Just curious, does [this sample](https://threejsfundamentals.org/threejs/threejs-responsive-hd-dpi.html) get a scrollbar as well on your machine?

**Comment by gkjohnson** [comment]
> @gkjohnson, Just curious, does this sample get a scrollbar as well on your machine?

That one _does not_ display the issue. I see that in your example `updateStyle` is passed as false to `setSize` and the width and height of the canvas are set to 100%. If setting `updateStyle` to false is an option then using `100vh` / `100vw` or `100%` for width and height fixes this issue, as well.

I do wonder if this is an OS windowing issue related to high dpi scaling considering it happens in all browsers and zooming in Chrome to a point where `devicePixelRatio` is fractional while the display scaling is a whole number does not cause the problem.

**Comment by greggman** [comment]
The issue is likely setting the canvas's css size explicitly since by the cascading rules the style applied to the element directly takes precedence. By setting the css to 100% or 100vw/vh you let the browser actually fill the window. By setting sized explicitly to `width: ???px; height: ???px;` like three.js does by default you try to override what the browser is doing. 

https://github.com/mrdoob/three.js/issues/4903

**Comment by gkjohnson** [comment]
> The issue is likely setting the canvas's css size explicitly since by the cascading rules the style applied to the element directly takes precedence.

Oh sure it's not a problem in most cases, though -- just on browsers on windows only when the OS high dpi settings are not whole numbers (granted I've only tested this laptop but it's happening in all browsers). It does seem like the two options are to bring "overflow: hidden" back or disable explicit setting of canvas styles in all the examples (or, dare I say it, use `!important` on the global canvas width and height parameters?!).

And FWIW my mac laptop works just fine when switching between all the different display scales.

**Comment by mrdoob** [comment]
@gkjohnson 

> I just realized this depends on the windows DPI scaling setting ([here](https://www.lifewire.com/using-windows-10-display-scaling-4587328)). I set my scaling to 125%. It sounds like yours is set to 200%. If I change my setting to 200% I can no longer replicate the problem and my `devicePixelRatio` is set to 2. See if you can replicate it with 125%? The issue doesn't always happen -- if the scrollbars don't show up try resizing the window and maybe refreshing.

I managed to replicate this once with a system-wide 125% scale and browser at 60%. Then I changed it to try another option and it never happened again.

I'm not going to put more efforts into this one unless someone figures out how to reproduce it consistently.
